core/src/main/scala/org/apache/spark/scheduler/SparkListenerBus.scala

```diff
@@ -24,7 +24,13 @@ import org.apache.spark.util.ListenerBus
  */
 private[spark] trait SparkListenerBus extends ListenerBus[SparkListener, SparkListenerEvent] {
 
+  private[spark] var filter: DefaultSparkListenerEventFilter = new DefaultSparkListenerEventFilter
+
   override def onPostEvent(listener: SparkListener, event: SparkListenerEvent): Unit = {
+    if (!filter.validate(event)) {
+      return  
```

Comment:
also, I think there is still one missing piece to get the json into the event logging for the history server.  You need to implement to put in the implementation for `onBlockUpdate` in `EventLoggingListener`.  I am suggesting that you just put this filter into that implementation, and get rid of the `Filter` abstraction.